code/modules/projectiles/gun.dm

```diff
@@ -609,6 +609,23 @@
 	update_gunlight()
 
 /obj/item/gun/proc/update_gunlight()
+	if(gun_light)
+		if(gun_light.on)
```

Comment:
```suggestion
		set_light_on(gun_light.on)
```

Reply:
Might have to add ``light_system = MOVABLE_LIGHT`` in ``/obj/item/gun``, as well as ``light_range`` and ``light_power``